pkg/controllers/federate/controller.go

```diff
@@ -519,6 +520,14 @@ func (c *FederateController) handleCreateFederatedObject(
 		return fmt.Errorf("failed to set pending controllers on federated object: %w", err)
 	}
 
+	if needsUpdate {
```

Comment:
Migration config has not been applied at this stage, it can be misleading to populate the annotation so soon. 
Consider adding the annotation after the scheduler has processed the migration config.